packages/notification-service/src/invites/invites.ts

```diff
@@ -7,27 +8,47 @@ import { getContractKit } from '../util/utils'
 
 const TAG = 'INVITES'
 
-export async function handleInvites() {
-  const kit = await getContractKit()
-  const fromBlock = getLastInviteBlockNotified() + 1
-  if (fromBlock <= 0) {
+let sendingInvites = false
+
+function updateLastInviteBlock(fromBlock: number, events: EventLog[]) {
+  if (events.length === 0) {
     return
   }
-  console.debug(TAG, `Starting to fetch invites from block ${fromBlock}`)
+  const maxBlock = events
+    .map((event) => event.blockNumber)
+    .reduce((max, current) => Math.max(max, current), fromBlock)
```

Comment:
Nit: I think this can all be expressed in a `reduce` so the `map` isn't necessary